### DISCUSSION TITLE
can textgrad read diffs?

### DISCUSSION THREAD

**Discussion by rawwerks** [zou-group/textgrad#30] [Q&A]
can textgrad read diffs? i think this would be particularly useful to applying textgrad to coding agents, which are often operating on files that are too large to fit into the context window. 

(bonus points if this can be integrated with git, so the teacher LLM is evaluating git commits of the student.)

**Comment by mertyg** [comment]
Great idea! Quick digression: I recently saw [this paper](https://arxiv.org/abs/2312.07540) doing this for agent optimization.
Need to think a bit about how to implement this. This feels akin to the current momentum implementation, where instead of keeping the latest k states of the variable, keep the diffs or something?

**Reply by rawwerks** [reply]
https://github.com/paul-gauthier/aider/ is the best example i can think of. but only "smart" LLMs can operate in "diff mode" (dumb ones need to keep the whole file in context).
